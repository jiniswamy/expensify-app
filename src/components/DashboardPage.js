import React from 'react';


const DashboardPage = () => (
  <div>
    Dashhboard Page content
  </div>
);

export default DashboardPage;
