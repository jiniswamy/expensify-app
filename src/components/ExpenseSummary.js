import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
export const ExpenseSummary = ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount == 1 ? 'expense' : 'expenses';
    const totalText = numeral(expenseTotal / 100).format('$0,0.00');

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span >{expenseCount}</span> {expenseWord} totaling <span>{totalText}</span>
                    <div className="page_header_action">
                        <Link className="button" to="/create">Add Expense</Link>
                    </div>
                </h1>
            </div>
        </div>
    );
};


const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: getExpensesTotal(state.expenses)
    };
};


export default connect(mapStateToProps)(ExpenseSummary);