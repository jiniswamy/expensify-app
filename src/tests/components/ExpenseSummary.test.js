import React from 'react'
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary'
import expenses from '../fixtures/expense';
import getExpensesTotal from '../../selectors/expenses-total';

test('should render ExpenseSummary with more expenses', () => {   
    const total = getExpensesTotal(expenses);   
    const wrapper = shallow(<ExpenseSummary expenses={expenses} expenseTotal={total} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with only one expenses',() => {
    const total = getExpensesTotal(expenses[0]); 
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]} expenseTotal={total} />);
    expect(wrapper).toMatchSnapshot();
})