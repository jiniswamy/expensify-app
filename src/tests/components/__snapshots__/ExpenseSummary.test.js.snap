// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with more expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totaling 
      <span>
        $1,141.95
      </span>
      <div
        className="page_header_action"
      >
        <Link
          className="button"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </h1>
  </div>
</div>
`;

exports[`should render ExpenseSummary with only one expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totaling 
      <span>
        $1.95
      </span>
      <div
        className="page_header_action"
      >
        <Link
          className="button"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </h1>
  </div>
</div>
`;
